File: cpp/2966.divide-array-into-arrays-with-max-difference.cpp
#include <vector>
using std::vector;

class Solution {
public:
  vector<vector<int>> divideArray(vector<int> &nums, int k) {
    const size_t n = nums.size();

    std::sort(nums.begin(), nums.end());
    vector<vector<int>> res;
    res.reserve(n / 3);

    for (int i = 0; i < n; i += 3) {
      if (nums[i + 2] - nums[i] > k) {
        return {};
      }

      res.push_back({nums[i], nums[i + 1], nums[i + 2]});
    }

    return res;
  }
};

File: golang/2966.divide-array-into-arrays-with-max-difference.go
func divideArray(nums []int, k int) [][]int {
    sort.Ints(nums)
    n := len(nums)
    res := make([][]int, 0)

    for i := 0; i < n; i += 3 {
        if nums[i + 2] - nums[i] > k {
            return [][]int{}
        } 
        res = append(res, nums[i: i + 3])
    }

    return res
}